### ISSUE TITLE
Doesn't work on certain link preview element anymore

### ISSUE THREAD

**Issue by PeterDaveHello** [Mte90/facebook-direct-links#39]
Look like there's problem to get the direct link from the link preview area.

I haven't digged it very deep yet, but use developer tool to take a look at the link preview area, found something interesting that the link will be changed to `https://l.facebook.com/l.php?u=` immediately when I clicked the right button, and changed back when the menu is closed or the mouse moved to some other places, not sure if it's done by Facebook though.

**Comment by Mte90** [comment]
I don't understand, what is the link preview area?

**Comment by PeterDaveHello** [comment]
Sorry not being clear enough. I mean the area to display link preview, like something you can see here when you put a url in the debug tool https://developers.facebook.com/tools/debug/, that Facebook will show preview of shared links.

**Comment by AntonioRedondo** [comment]
Yeah, I'm experiencing the same issue. The link cleaning isn't working anymore.

This happens on posts of pages on the big preview created when sharing external links on a post. Previously, with the add-on installed, when clicking the preview link the browser would go straight to the final page without `https://l.facebook.com/l.php?u=`. Now this cleaning isn't happening anymore and the nasty Facebook tracking redirection is active again, as if Facebook Direct Links wasn't installed.

**Comment by Mte90** [comment]
So with big preview do you mean the links when you are looking a photo on facebook at fullsize with the sidebar on the right?
What links are not working in that view? without an example link is not easy to understand where it is happening the issue.

**Comment by AntonioRedondo** [comment]
It stopped working for all preview links. For example:
https://www.facebook.com/bbcnews/posts/pfbid0qFD6Bm8aazoGkDyjKQP1uVbXs6MUqQ92WRQtVY9tGqueMXV8HfYK2U3DuUSHboGPl

![image](https://user-images.githubusercontent.com/5444421/194569119-4a3c7e79-0a4a-49f3-8b4a-b889573d4a3a.png)

Please note that when hovering over the link you can see that it isn't _facebooked_ yet. But when you click it it actually goes through `https://l.facebook.com/l.php?u=`.

**Comment by Mte90** [comment]
I tested it and on my case I don't get that URL.

**Comment by AntonioRedondo** [comment]
This is happening to me on two different Windows 11 computers with Firefox 105 and Facebook Direct Links 1.3.6.

**Comment by rfc1036** [comment]
It has been broken for me as well for a while.

**Comment by Mte90** [comment]
I tested right now and I get:

![Screenshot_20230327_094558](https://user-images.githubusercontent.com/403283/227874966-5ccec4c5-82ea-443a-837d-8cb03d97294a.png)

After a click:
![immagine](https://user-images.githubusercontent.com/403283/227875073-e905fac5-f448-4790-99c3-973394ec10c1.png)

Moving the mouse on the photo automatically change the link to the clean one

**Comment by Mte90** [comment]
I am investigating and seems that facebook just on hover change the URL with the one of the link. 
Initially is a encrypted version like:
![Screenshot_20230402_145314](https://user-images.githubusercontent.com/403283/229354120-96c291f3-7858-4cdb-8a4b-e241edc9847c.png)
On hover change the URL of the link with the direct one:
![immagine](https://user-images.githubusercontent.com/403283/229354204-9da45430-8084-40e7-bfc8-03055de0507d.png)

So we can't do anymore the trick of the extension as the URL was in the HTML page and extract it, now the URL is tracked server side and generated (probably for GDPR).
I will close the project and remove it from the addons store in the next days as it is clearly a feature that as browser extension we can't offer anymore as everything is crypted and parsed server side.

**Comment by PeterDaveHello** [comment]
If it's a direct link in the post, instead of the "preview" link, this extension is still working, maybe don't need to drop it?

**Comment by Mte90** [comment]
There isn't any direct link in the post, the link is generated by a AJAX request on hover in the box. in the `href="#"` section above.

**Comment by PeterDaveHello** [comment]
@Mte90 I mean not those one mentioned in the subject of this issue, like this one:

![image](https://user-images.githubusercontent.com/3691490/229356001-5828933e-b2c6-4d96-a0da-057c8cff9d4d.png)

We can still copy the real link without problems, the link is https://beta.openai.com/ (from https://www.facebook.com/openai.research)

**Comment by PeterDaveHello** [comment]
What I'm trying to say if not all the features are dead, maybe you don't need to close the project and remove it from the addons store?

**Comment by Mte90** [comment]
Well it is most used for the big preview images with links, the cases with a URL in a fb post are very low.
Yes the extension in those cases remove the tracking stuff as it is still using the old way to generate those link and our trick works, but how much before they change also that?

I mean I had to search a lot to find a post with just a link as now everyone prefer the preview link with the image as it is more useful for engagement and get a click.

**Comment by PeterDaveHello** [comment]
Not sure, but delete it will make sure we have no chance to bypass any of them, keeping it will at least let us to bypass some of them, even not so much.

**Comment by Mte90** [comment]
The problem is still not working for all the links but users expect that. So this require some changes on the description as example but I don't want to get opened various tickets asking why on preview links is not working.

**Comment by PeterDaveHello** [comment]
@Mte90 got your point. Maybe get pin this issue first? So people will see it first.

**Comment by Mte90** [comment]
Pinned the issue but I think that is not enough we need to something on the communications side for this.

**Comment by Mte90** [comment]
So I added a note in the Readme and in the Addons.Mozilla.org description about the Facebook changes.

I think that there isn't else we can do about it sadly.

**Comment by PeterDaveHello** [comment]
Not sure if this idea would be too ideal, what about we try to send the request to the server to get the decrypted URL in the background? Need to test the latency though.

**Comment by Mte90** [comment]
Well facebook already do that decryption when you click on a link, how I didn't investigated but I don't think that is easy to understand the process.
I just saw like 4/5 different server request.
